test/common/http/utility_test.cc

```diff
@@ -137,4 +137,21 @@ TEST(HttpUtility, TestParseCookieWithQuotes) {
   EXPECT_EQ(Utility::parseCookieValue(headers, "leadingdquote"), "\"foobar");
 }
 
+TEST(HttpUtility, TestRemoveCookieValues) {
+  TestHeaderMapImpl headers{
+      {"someheader", "10.0.0.1"},
+      {"cookie", "foo=bar"},
+      {"cookie", "boo=yah; foo=bar"},
+      {"cookie", "foo=baz; bananas;"},
+      {"cookie", "abc=def"}};
+
+  Utility::removeCookieValues(headers, "foo")
+
+  TestHeaderMapImpl expected{
+      {"someheader", "10.0.0.1"},
+      {"cookie", "abc=def"}};
+
```

Comment:
I guess this partly answers the question above, i.e. you want to remove the entire cookie header. Question is whether this is valid or not. And finally, a use case for this would be helpful, because we could add this to the document.